sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -189,17 +189,22 @@ object SubExprUtils extends PredicateHelper {
    */
   def getOuterReferences(expr: Expression): Seq[Expression] = {
     val outerExpressions = ArrayBuffer.empty[Expression]
```

Comment:
@cloud-fan . This is a core logic refactoring instead of `Improve error message for correlated subquery`.
Could you revise the PR title and description?

Reply:
done